src/main/java/io/confluent/connect/elasticsearch/ElasticsearchWriter.java

```diff
@@ -220,7 +238,19 @@ public void write(Collection<SinkRecord> records) {
       final IndexableRecord indexableRecord = converter.convertRecord(sinkRecord, index, type,
                                                                       ignoreKey, ignoreSchema);
 
-      bulkProcessor.add(indexableRecord, flushTimeoutMs);
+      // In the event that the sink record's value was null and the data converter has been told to
+      // ignore null values, the returned record will be null.
```

Comment:
Had a `TODO` here just in case it turned out it might be advantageous for the `ElasticsearchWriter` to intercept null-valued records before this point when `behavior.on.null.values` is set to `IGNORE`, in order to avoid possible performance penalties and/or other issues arising from the `if (!ignoreSchema && !existingMappings.contains(index))` block. Should that be put back in or was it removed intentionally?

Reply:
I removed it intentionally. We can change the code now, but I don't think the speculative TODO was useful.

Reply:
Alright, fair enough. Guess it can't hurt to change it now--as conceptually pleasing as it is to [dr]elegate (😜) handling of null-valued records to the `DataConverter` class I see some calls to the `JestClient` inside the for-loop above that would probably be beneficial to avoid.

Reply:
Let me know if you have problems pushing to this branch.